src/nylas-connection.ts

```diff
@@ -316,8 +329,15 @@ export default class NylasConnection {
           }
         })
         .catch((err: Error) => {
+          if (err && err.name && err.name === 'AbortError') {
+            console.warn('Request timed out');
+            return reject(err);
+          }
           console.error(`Error encountered during request:\n${err.stack}`);
           return reject(err);
+        })
+        .then(() => {
```

Comment:
```suggestion
        .finally(() => {
```

This is to clear the timeout regardless of what happens, right (e.g. success or timeout)?